README.md

```diff
@@ -16,17 +16,63 @@
+- The algorithms may require a lot of experimentation before one gets them right. If so, the iteration time required to implement a change, deploy it, and see the result should be in single minutes if not seconds. With Nussknacker, unless the change is significant, non-technical users can achieve iteration time below one minute.
+- To support high productivity, low-code tools must provide features developers use in professional IDEs - Nussknacker Designer has built-in [syntax checking, code completion](https://nussknacker.io/documentation/about/KeyFeatures/#smart-code-suggestions-and-validation), [debugging, and testing support](https://nussknacker.io/documentation/docs/next/scenarios_authoring/TestingAndDebugging/).
+- Experimenting with algorithms requires observability going beyond pure technical metrics like throughput, Kafka topics lag, etc. Out of the box, Nussknacker comes with an integrated and ready-to-use monitoring subsystem which allows to monitor not only technical aspects of the working algorithm but also its internal behavior - for example [events count](https://nussknacker.io/documentation/docs/next/scenarios_authoring/TestingAndDebugging/#watching-nodes-filtering-behaviour-with-counts) per algorithm node. You will not need to spend developers' time on this functionality.
+- Last but not least, architecture matters. Nussknacker can deliver very high throughput, horizontal scalability, resilience, and high availability through the use of tools and platforms known for their rock-solid architecture - Kafka, Flink, and Kubernetes.
+
+
+Check out [this document](https://nussknacker.io/documentation/about/KeyFeatures/) for a concise summary of Nussknacker features. 
+
+
+## Use cases
+
+![image](./docs/NuHighLevelView.png)
+
+Nussknacker is typically used as a component of a larger system; rarely it is an end-to-end solution. The use cases follow a common pattern: a program working on a data stream, file or in a request-response interaction style receives a set of data (event or request) and has to deliver a decision. To “take” the decision it needs to perform one or more of the following: discard irrelevant records, enrich incoming records with data from external sources, aggregate events in time windows (if working on a data stream), run one or more ML models, compute the decision and finally deliver it either as another data stream, file or a response.  The ‘decisions’ can be from a broad spectrum of vertical and horizontal applications:
```

Comment:
1. WDYT: `a component of a larger system; rarely it is an end-to-end solution.` -> `a component of a larger system but it can be used as an end-to-end solution too.`

2. missing comma: `request-response interaction style receives a set ` -> `request-response interaction style, receives a set `

3. `deliver a decision` ... hmm, I'm not sure if it fits well streaming use case

Reply:
1. applied
2. I think there should not be a comma after last item of a list.
3. No idea how to formulate it in a better way.

Reply:
Re 2. reapplied comma.
Re 3. After off line discussion, we agreed to stick to decisions.